src/login/AzureLoginHelper.ts

```diff
@@ -189,27 +190,27 @@ export class AzureAccountLoginHelper {
 		});
 	}
 
-	private async loadSubscriptionCache(): Promise<void> {
-		const cache: ISubscriptionCache | undefined = this.context.globalState.get(cacheKey);
+	private async loadSubscriptionTenantCache(): Promise<void> {
+		const cache: SubscriptionTenantCache | undefined = this.context.globalState.get(cacheKey);
 		if (cache) {
 			(<IAzureAccountWriteable>this.api).status = 'LoggedIn';
 			const sessions: Record<string, AzureSession> = await this.authProvider.initializeSessions(cache, this.api);
 
-			const subscriptions: AzureSubscriptionInternal[] = cache.subscriptions.map<AzureSubscriptionInternal>(({ session, subscription, tenants }) => {
+			const subscriptions: AzureSubscription[] = cache.subscriptions.map<AzureSubscription>(({ session, subscription }) => {
 				const { environment, userId, tenantId } = session;
 				const key: string = getKey(environment, userId, tenantId);
 				return {
 					session: sessions[key],
-					subscription,
-					tenants
+					subscription
 				};
 			});
 			this.subscriptionsTask = Promise.resolve(subscriptions);
 			this.api.subscriptions.push(...subscriptions);
+			this.tenantsTask = Promise.resolve(cache.tenants);
 
 			const resourceFilter: string[] | undefined = getSettingValue(resourceFilterSetting);
 			this.oldResourceFilter = JSON.stringify(resourceFilter);
-			const newFilters: AzureResourceFilterInternal[] = getNewFilters(subscriptions, resourceFilter);
+			const newFilters: AzureSubscription[] = getNewFilters(subscriptions, resourceFilter);
```

Comment:
Is this supposed to be `AzureSubscription[]`? The return type of `getNewFilters` is `AzureResourceFilter[]`.  Does `AzureSubscription` extend it?

Reply:
Yessir they're actually the same type:

https://github.com/microsoft/vscode-azure-account/blob/9b8a76ddc8f9ce181f71fdf283b1d23009ca37c2/src/azure-account.api.d.ts#L47

This as well as the other instance you mentioned should technically be `AzureResourceFilter` but didn't wanna crowd the PR with those changes.

Reply:
Took care of this here: https://github.com/microsoft/vscode-azure-account/pull/400 💯💯💯💯